native/src/main/java/io/ballerina/stdlib/data/utils/DataUtils.java

```diff
@@ -0,0 +1,735 @@
+            }
+        }
+        return key;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static String processNameAnnotation(BMap<BString, Object> annotation, String key, BString value,
+                                                boolean hasNamespaceAnnotation) {
+        String nameValue = ((BMap<BString, Object>) annotation.get(value)).
+                get(StringUtils.fromString(VALUE)).toString();
+        if (hasNamespaceAnnotation) {
+            return key.substring(0, key.indexOf(Constants.COLON) + 1) + nameValue;
+        } else {
+            return nameValue;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static String processNamespaceAnnotation(BMap<BString, Object> annotation, String key, BString value,
+                                                     BMap<BString, Object>  subRecord) {
+        BMap<BString, Object> namespaceAnnotation = (BMap<BString, Object>) annotation.get(value);
+        BString uri = (BString) namespaceAnnotation.get(Constants.URI);
+        BString prefix = (BString) namespaceAnnotation.get(Constants.PREFIX);
+        if (prefix ==  null) {
+            subRecord.put(StringUtils.fromString(ATTRIBUTE_PREFIX + "xmlns"), uri);
+        } else {
+            subRecord.put(StringUtils.fromString(ATTRIBUTE_PREFIX + "xmlns:" + prefix), uri);
+            key = prefix.getValue().concat(Constants.COLON).concat(key);
+        }
+        return key;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static QName processFieldNamespaceAnnotation(BMap<BString, Object> annotation, String key, BString value,
+                                                          BMap<BString, Object>  subRecord, boolean isAttributeField) {
+        BMap<BString, Object> namespaceAnnotation = (BMap<BString, Object>) annotation.get(value);
+        BString uri = (BString) namespaceAnnotation.get(Constants.URI);
+        BString prefix = (BString) namespaceAnnotation.get(Constants.PREFIX);
+        QName qName = new QName(uri == null ? "" : uri.getValue(), key, prefix == null ? "" : prefix.getValue());
+        if (isAttributeField) {
+            addAttributeToRecord(prefix, uri, key, subRecord);
+        } else {
+            BMap<BString, Object> nextMapValue = ValueCreator.createMapValue(Constants.JSON_MAP_TYPE);
+            key = addAttributeToRecord(prefix, uri, key, nextMapValue);
+            subRecord.put(StringUtils.fromString(key), nextMapValue);
+        }
+        return qName;
+    }
+
+    private static String addAttributeToRecord(BString prefix, BString uri, String key,
+                                               BMap<BString, Object> subRecord) {
+        if (prefix ==  null) {
+            subRecord.put(StringUtils.fromString(ATTRIBUTE_PREFIX + "xmlns"), uri);
+        } else {
+            subRecord.put(StringUtils.fromString(ATTRIBUTE_PREFIX + "xmlns:" + prefix), uri);
+            key = prefix.getValue().concat(Constants.COLON).concat(key);
+        }
```

Comment:
If we do the check the other way round, we can return without modifying `key` which is a parameter, please check other places too.